internal/broker/config_test.go

```diff
@@ -150,3 +152,216 @@ func TestParseConfig(t *testing.T) {
+			wantFirstUserBecomesOwner: false,
+			wantOwner:                 "machine_owner",
+			wantAllowedUsers:          []string{},
+		},
+		"By_default_only_owner_is_allowed": {
+			wantAllUsersAllowed:       false,
+			wantOwnerAllowed:          true,
+			wantFirstUserBecomesOwner: false,
+			wantOwner:                 "machine_owner",
+			wantAllowedUsers:          []string{},
+		},
+		"Only_owner_is_allowed_but_is_unset": {
+			wantAllUsersAllowed:       false,
+			wantOwnerAllowed:          true,
+			wantFirstUserBecomesOwner: true,
+			wantOwner:                 "",
+			wantAllowedUsers:          []string{},
+		},
+		"Only_owner_is_allowed_but_is_empty": {
+			wantAllUsersAllowed:       false,
+			wantOwnerAllowed:          true,
+			wantFirstUserBecomesOwner: false,
+			wantOwner:                 "",
+			wantAllowedUsers:          []string{},
+		},
+		"Users_u1_and_u2_are_allowed": {
+			wantAllUsersAllowed:       false,
+			wantOwnerAllowed:          false,
+			wantFirstUserBecomesOwner: false,
+			wantOwner:                 "",
+			wantAllowedUsers:          []string{"u1", "u2"},
+		},
+		"Unset_owner_and_u1_is_allowed": {
+			wantAllUsersAllowed:       false,
+			wantOwnerAllowed:          true,
+			wantFirstUserBecomesOwner: true,
+			wantOwner:                 "",
+			wantAllowedUsers:          []string{"u1"},
+		},
+		"Set_owner_and_u1_is_allowed": {
+			wantAllUsersAllowed:       false,
+			wantOwnerAllowed:          true,
+			wantFirstUserBecomesOwner: false,
+			wantOwner:                 "machine_owner",
+			wantAllowedUsers:          []string{"u1"},
+		},
+	}
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			outDir := t.TempDir()
+			confPath := filepath.Join(outDir, "broker.conf")
+
+			err := os.WriteFile(confPath, []byte(testParseUserConfigTypes[name]), 0600)
+			require.NoError(t, err, "Setup: Failed to write config file")
+
+			dropInDir := confPath + ".d"
+			err = os.Mkdir(dropInDir, 0700)
+			require.NoError(t, err, "Setup: Failed to create drop-in directory")
+
+			cfg, err := parseConfigFile(confPath, p)
+
+			// convert the allowed users array to a map
+			allowedUsersMap := map[string]struct{}{}
+			for _, k := range tc.wantAllowedUsers {
+				allowedUsersMap[k] = struct{}{}
+			}
+
+			require.Equal(t, tc.wantAllUsersAllowed, cfg.allUsersAllowed)
+			require.Equal(t, tc.wantOwnerAllowed, cfg.ownerAllowed)
+			require.Equal(t, tc.wantOwner, cfg.owner)
+			require.Equal(t, tc.wantFirstUserBecomesOwner, cfg.firstUserBecomesOwner)
+			require.Equal(t, allowedUsersMap, cfg.allowedUsers)
+
+			require.NoError(t, err)
+			golden.CheckOrUpdateFileTree(t, outDir)
+		})
+	}
+}
+
+func TestRegisterOwner(t *testing.T) {
+	outDir := t.TempDir()
+	userName := "owner_name"
+	confPath := filepath.Join(outDir, "broker.conf")
+
+	err := os.WriteFile(confPath, []byte(configTypes["valid"]), 0600)
+	require.NoError(t, err, "Setup: Failed to write config file")
+
+	dropInDir := confPath + ".d"
```

Comment:
Can reuse `GetDropInDir`?

Reply:
So I just want to offer my logic behind this, in tests IMO in tests it's better not to use helper functions or constants (e.g. a couple of lines we use `"20-owner-autoregistration.conf"` to generate the path of the file, instead of using the const `ownerAutoRegistrationConfigPath`). The reason for this is to act as a safe guard for typos in the code and to make changing this behavior harder. Also this function is used in the function that we are trying to test, so from a higher level I don't like this.

Of course, this does not always make sense and there are cases where it makes changes to the code harder (but I think that behavior like this is unlikely to change, so this is not the case).

Anyway, this is a nit-pick I would say, I don't think there is any real difference between the 2 approaches and I understand the reason, so I will update it.